### PR TITLE
Fix the istioctl x ps is missing ztunnel statuses 

### PR DESCRIPTION
**Pull request by yankay**
**Please provide a description of this PR:**

Fix https://github.com/istio/istio/issues/43877

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ X] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by hanxiaop — APPROVED** [review]

**Review by GregHanson — COMMENTED** [review]
```
istioctl ps
NAME                                                   CLUSTER        CDS          LDS          EDS          RDS          ECDS         ISTIOD                      VERSION
ztunnel-gpl9c.istio-system                                            NOT SENT     NOT SENT     NOT SENT     NOT SENT     NOT SENT     istiod-5b979677cd-dmdqw     1.18-dev
ztunnel-hgfsw.istio-system                                            NOT SENT     NOT SENT     NOT SENT     NOT SENT     NOT SENT     istiod-5b979677cd-dmdqw     1.18-dev
ztunnel-kfv6c.istio-system                                            NOT SENT     NOT SENT     NOT SENT     NOT SENT     NOT SENT     istiod-5b979677cd-dmdqw     1.18-dev
```
I'm not sure if `NOT SENT` is the right status to report for ztunnel, since it will never receive a LDS/CDS/RDS/EDS/ECDS event, only WDS.  Those other status are reserved for envoy based proxies

File: pilot/pkg/xds/statusgen.go [hunk 1]
```diff
@@ -106,6 +107,11 @@ func isProxy(con *Connection) bool {
 		con.proxy.Metadata.ProxyConfig != nil
 }
 
+func isZtunnel(con *Connection) bool {
+	return con != nil && con.proxy != nil &&
+		con.proxy.Metadata != nil && strings.HasPrefix(con.proxy.ID, "ztunnel")
```
Comment (howardjohn) [inline pilot/pkg/xds/statusgen.go:112]: I would check con.proxy.Type

**Comment by hanxiaop** [comment]
> For instance, ingress gateway will show `NOT SENT` for its RDS until a VirtualService is created and matched against it. It doesn't have one yet, but it _can_ receive one eventually

How about introducing a new status like `IGNORED` to the ps command to indicate the status will never be updated to `SYNCED` status?

File: pilot/pkg/xds/statusgen.go (hunk 1 above)
Comment (yankay) [reply pilot/pkg/xds/statusgen.go]: Thanks @hanxiaop , it has been changed. :-)

**Comment by yankay** [comment]
Hi @GregHanson ,
The PR has been changed follow the advice , would you please review it again. :-)

**Review by GregHanson — APPROVED** [review]
new output:
```
istioctl ps
NAME                                                   CLUSTER        CDS         LDS         EDS         RDS          ECDS         ISTIOD                      VERSION
ztunnel-6tscn.istio-system                                            IGNORED     IGNORED     IGNORED     IGNORED      IGNORED      istiod-8454d74fcc-rfxck     1.18-dev
ztunnel-7ppv5.istio-system                                            IGNORED     IGNORED     IGNORED     IGNORED      IGNORED      istiod-8454d74fcc-rfxck     1.18-dev
ztunnel-twhbt.istio-system                                            IGNORED     IGNORED     IGNORED     IGNORED      IGNORED      istiod-8454d74fcc-rfxck     1.18-dev
istio-ingressgateway-786db5cdff-6glpq.istio-system     Kubernetes     SYNCED      SYNCED      SYNCED      NOT SENT     NOT SENT     istiod-8454d74fcc-rfxck     1.18-dev
```

LGTM

**Comment by yankay** [comment]
Thanks @hanxiaop @GregHanson @howardjohn 
for the PR review :-)

**Pull request merged**